### ISSUE TITLE
Unable to SSH into EC2 Instance

### ISSUE THREAD

**Issue by ajmeese7** [openemr/openemr#5087]
### Describe the bug

I spun up an OpenEMR stack and for some strange reason I am unable to SSH into the EC2 instance no matter what I do. The security policy clearly allows port 22 inbound and any traffic outbound, but when I run the suggested ` ssh -i "pemKeyNameHere.pem" ubuntu@ec2-my-ip-here-.compute-1.amazonaws.com` I always get `Permission denied (publickey)`. I have already done `chmod 400 pemName.pem` so the accessibility of the key is not an issue, but the server is still refusing any requests to SSH into it.

Additional question, how do I modify the code of OpenEMR on the instance while it is running? Even if I am able to get SSH functioning, the code is assumably isolated in a Docker instance. I would like to make my own code modifications to the instance and I would like to know if this is possible to do on the server or if I need to make the changes on my local machine then create my own Docker image to push to the server.

Thank you! :)

### Client configuration

- Browser: Brave
- OpenEMR version: OpenEMR Express Plus 6.1.0
- Operating system: Windows 11 (using Ubuntu WSL2 for the SSH)

**Comment by jesdynf** [comment]
Hi,

Are you able to connect to other EC2 instances with the same procedure? I suspect this is not an OpenEMR issue.

**Comment by adunsulag** [comment]
@ajmeese7 as far as modifying the data on the docker machines, look at our contributing guide here for working with docker & OpenEMR: https://github.com/openemr/openemr/blob/master/CONTRIBUTING.md

**Comment by ajmeese7** [comment]
@jesdynf yes I am, I have another EC2 instance running Ubuntu currently and I can connect to it with no problems at all. Both use the same PEM, the only difference is that I used the OpenEMR configuration for this current stack and I used the Ubuntu AMI for my other instance.

@adunsulag I did read that document, and I understand that it is recommended to use local development with Docker to push code changes to the main repository. What I am interested in is whether or not it is possible to make live modifications to the stack spun up by the OpenEMS configuration file, I have never tried to manipulate a Docker instance's code in real-time and I am curious if anybody here has tried it before.

Thank you both for responding!

**Comment by jesdynf** [comment]
It sounds like you started an OpenEMR Standard stack? See if you have the same problem with OpenEMR Express -- just launch it, try to connect, and shut it down after.

**Comment by adunsulag** [comment]
@ajmeese7 The docker filesystem is ephemeral unless you have the codebase mounted as a shared volume (which is the setup with the development-easy docker setup).  Any live changes you make to the docker unless you generate a new docker image will not persist if the docker is ever shut down or restarted.  

If you want to run on the cloud and make live changes I recommend just starting up your own AWS ami and then running it like a local box using the development-easy docker-compose file.  I've done that when I've needed to make custom changes that are accessible to clients / customers.  

If you try to modify the docker files in our standard/express install I think you'll run into problems as that goes against the way most production dockers are setup and run.

**Comment by jesdynf** [comment]
Well, the changes won't persist if the container is rebuilt, but the containers can be stopped or started just fine, as long as `docker-compose down` (or a corresponding `up`) is never called.

**Comment by ajmeese7** [comment]
> It sounds like you started an OpenEMR Standard stack? See if you have the same problem with OpenEMR Express -- just launch it, try to connect, and shut it down after.

I started an OpenEMR Express Plus stack, as mentioned in the initial issue. I haven't tried a standard stack though, but I'm not sure if that would make a difference.

@adunsulag that's a great idea, thanks! I just need to pray that my custom AMI will allow me to SSH into it 😆 

Thank you both! I'll leave this issue open for now in hopes that someone can give guidance on the problem at hand, but I'm probably going to pursue the solutions that you two suggested, so I'll close the issue out in a few days if no one can offer a different idea of what the root problem might be.

**Comment by jesdynf** [comment]
Express *Plus* won't let you log in? That's just mad. It's /literally/ the Ubuntu 20.04 LTS instance in your region, that's what I used as a base, it hasn't even gone through the account stripping its siblings have.

Although... did you actually select your EC2 key when you launched the template? If you left that field blank you wouldn't be able to connect at all.

**Comment by ajmeese7** [comment]
@jesdynf I don't remember manully selecting my EC2 key, but it automatically selected the primary one for the organization that I am currently contracting for.

![EC2KeyPair Parameter](https://user-images.githubusercontent.com/17814535/161644774-3f0430ab-1608-4e14-831b-a17a3faefeda.png)

I completely agree with you, this one has me stumped too. I've never had this problem with AWS before, which is what led me to assume that there was some kind of an issue either with the Express configuration or with something that I did wrong on my end with OpenEMR.

**Comment by adunsulag** [comment]
@ajmeese7 If you launch a stock vanilla Ubuntu instance and you can't SSH into it then that's not an OpenEMR issue.  You'd want to look if the organization has any IAM policies that are restricting what key / AMIs you can use.  There could be some other cloud governance rules they've setup that could be restricting you also.

**Comment by jesdynf** [comment]
Well, is SleepFromHome the key you're presenting from the command line -- is that the key you built your test instance with? I was testing Express Plus just yesterday, I know it works fine.

**Comment by ajmeese7** [comment]
@adunsulag @jesdynf somehow there is something wrong with my new machine! For some reason unbeknownst to me, all of a sudden I cannot connect to any AWS instances via SSH. I tried from the Windows command line, my WSL2 Ubuntu distro, and the Git Bash CLI, and no dice. The same error across all of them, including on a brand new Ubuntu 20.04 LTS instance! I have no clue what the problem could be but I am confident now that you two are correct, whatever is going on with my machine has nothing to do with the configuration of OpenEMR.

I know the organization doesn't have any additional cloud restrictions because I created the organization on behalf of my employer, I am the only person who has interacted with the account thus far. This is really weird!

Thank you so much for your time, I'm sorry for wasting it. I'm going to go try to unbrick whatever happened to my machine, wish me luck!
